src/borg/archiver.py

```diff
@@ -1147,13 +1147,8 @@ def do_umount(self, args):
     @with_repository(compatibility=(Manifest.Operation.READ,))
     def do_list(self, args, repository, manifest, key):
         """List archive or repository contents"""
-        if not hasattr(sys.stdout, 'buffer'):
-            # This is a shim for supporting unit tests replacing sys.stdout with e.g. StringIO,
-            # which doesn't have an underlying buffer (= lower file object).
-            def write(bytestring):
-                sys.stdout.write(bytestring.decode('utf-8', errors='replace'))
-        else:
-            write = sys.stdout.buffer.write
+        def write(bytestring):
+            sys.stdout.write(bytestring.decode('utf-8', errors='replace'))
```

Comment:
so, in the end it first calls safe_encode (utf-8 encode, surrogate-escape) and then decode (utf-8, replace).

guess we could just call remove_surrogates (utf-8 encode, replace + utf-8 decode) - like at a lot of other places.